Framework/DataHandling/src/LoadDNSEvent.cpp

```diff
@@ -0,0 +1,525 @@
+  }
+
+  // combine neutronEvents:
+  PARALLEL_FOR_NO_WSP_CHECK()
+  for (int i = 0; i < static_cast<int>(finalEventAccumulator.neutronEvents.size()); ++i) {
+    auto &allNeutronEvents = finalEventAccumulator.neutronEvents[static_cast<size_t>(i)];
+    for (const auto &v : eventAccumulators) {
+      allNeutronEvents.insert(allNeutronEvents.end(), v.neutronEvents[static_cast<size_t>(i)].begin(),
+                              v.neutronEvents[static_cast<size_t>(i)].end());
+    }
+  }
+
+  // parse_BlockList(file, _eventAccumulator);
+  // parse_EndSignature(file);
+
+  return finalEventAccumulator;
+}
+
+void LoadDNSEvent::parse_BlockList(VectorByteStream &file, EventAccumulator &eventAccumulator) {
+  // BlockList := DataBuffer BlockListTrail
+  while (!file.eof() && file.peek() != 0xFF) {
+    parse_Block(file, eventAccumulator);
+  }
+}
+
+void LoadDNSEvent::parse_Block(VectorByteStream &file, EventAccumulator &eventAccumulator) {
+  // Block := DataBufferHeader DataBuffer
+  parse_DataBuffer(file, eventAccumulator);
+  parse_BlockSeparator(file);
+}
+
+void LoadDNSEvent::parse_BlockSeparator(VectorByteStream &file) {
+  auto separator = file.readRaw(separator_t());
+  if (separator != block_sep) {
+    throw std::runtime_error(std::string("File Integrety LOST. (ugh!) 0x") + n2hexstr(separator) +
+                             std::string("expected 0x") + n2hexstr(block_sep));
+  }
+}
+
+void LoadDNSEvent::parse_DataBuffer(VectorByteStream &file, EventAccumulator &eventAccumulator) {
+  const auto bufferHeader = parse_DataBufferHeader(file);
+
+  const uint16_t dataLength = uint16_t(bufferHeader.bufferLength - 21);
+  const auto event_count = dataLength / 3;
+
+  for (uint16_t i = 0; i < event_count; i++) {
+    parse_andAddEvent(file, bufferHeader, eventAccumulator);
+  }
+}
+
+LoadDNSEvent::BufferHeader LoadDNSEvent::parse_DataBufferHeader(VectorByteStream &file) {
+  uint16_t ts1;
+  uint16_t ts2;
+  uint16_t ts3;
+  BufferHeader header = {};
+  file.read<2>(header.bufferLength);
+  file.read<2>(header.bufferVersion);
+  file.read<2>(header.headerLength);
+  file.read<2>(header.bufferNumber);
+  file.read<2>(header.runId);
+  file.read<1>(header.mcpdId);
+  file.read<1>(header.deviceStatus);
+  file.read<2>(ts1);
+  file.read<2>(ts2);
+  file.read<2>(ts3);
+  // 48 bit timestamp is 3 2-byte MSB words ordered LSB
+  header.timestamp = (uint64_t)ts3 << 32 | (uint64_t)ts2 << 16 | ts1;
+  // g_log.debug() << int(header.bufferLength)  << " "
+  //              << int(header.headerLength)  << " "
+  //              << int(header.bufferNumber)  << " "
+  //             << int(header.runId)  << " "
+  //             << int(header.runId)  << " "
+  //             << int(header.mcpdId)  << " "
+  //             << int(header.deviceStatus)  << " "
+  //             << ts1  << " "
+  //             << ts2  << " "
+  //             << ts3  << " "
+  //             << header.timestamp  << " "
+  //             << std::endl;
```

Comment:
I removed it, while it might be useful for testing, I do not want to have it in the release.
Since if you set the output to debug and run it on a normal sized datafile, you will get prints to the console for hours.